reports/src/main/java/com/ivy/reports/ReportViewModel.kt

```diff
@@ -1,415 +1,418 @@
+import com.ivy.wallet.utils.scopedIOThread
+import com.ivy.wallet.utils.uiThread
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.async
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.launch
-import java.math.BigDecimal
+import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 @HiltViewModel
 class ReportViewModel @Inject constructor(
-    private val plannedPaymentsLogic: PlannedPaymentsLogic,
-    private val settingsDao: SettingsDao,
-    private val transactionDao: TransactionDao,
-    private val ivyContext: com.ivy.core.ui.temp.IvyWalletCtx,
-    private val nav: Navigation,
+    private val accountsAct: AccountsAct,
+    private val categoriesAct: CategoriesAct,
+    private val queryTransactionsAct: TrnsAct,
+    private val calculateAct: CalculateWithTransfersAct,
+    private val exchangeAct: ExchangeAct,
+    private val baseCurrencyAct: BaseCurrencyAct,
+    private val groupTransactionsAct: GroupTrnsAct,
     private val exportCSVLogic: ExportCSVLogic,
-    private val exchangeAct: ExchangeActOld,
-    private val accountsAct: AccountsActOld,
-    private val categoriesAct: CategoriesActOld,
-    private val trnsWithDateDivsAct: TrnsWithDateDivsAct,
-    private val calcTrnsIncomeExpenseAct: CalcTrnsIncomeExpenseAct,
-    private val baseCurrencyAct: BaseCurrencyActOld
-) : FRPViewModel<ReportScreenState, Nothing>() {
+    private val plannedPaymentsLogic: PlannedPaymentsLogic,
+) : FRPViewModel<ReportScreenState, ReportScreenEvent>() {
     override val _state: MutableStateFlow<ReportScreenState> = MutableStateFlow(
         ReportScreenState()
     )
 
-    override suspend fun handleEvent(event: Nothing): suspend () -> ReportScreenState {
-        TODO("Not yet implemented")
-    }
-
-    private val unSpecifiedCategory =
-        CategoryOld(com.ivy.core.ui.temp.stringRes(R.string.unspecified), color = Gray.toArgb())
-
-    private val _period = MutableLiveData<TimePeriod>()
-    val period = _period.asLiveData()
-
-    private val _categories = MutableStateFlow<List<CategoryOld>>(emptyList())
-    val categories = _categories.readOnly()
-
-    private val _allAccounts = MutableStateFlow<List<AccountOld>>(emptyList())
-
     private val _baseCurrency = MutableStateFlow("")
-    val baseCurrency = _baseCurrency.readOnly()
-
-    private val _historyIncomeExpense = MutableStateFlow(IncomeExpenseTransferPair.zero())
-    private val historyIncomeExpense = _historyIncomeExpense.readOnly()
-
     private val _filter = MutableStateFlow<ReportFilter?>(null)
-    val filter = _filter.readOnly()
-
-    fun start() {
-        viewModelScope.launch(Dispatchers.IO) {
-            _baseCurrency.value = baseCurrencyAct(Unit)
-            _allAccounts.value = accountsAct(Unit)
-            _categories.value = listOf(unSpecifiedCategory) + categoriesAct(Unit)
+    private val allAcc = MutableStateFlow<List<Account>>(emptyList())
```

Comment:
We don't need those `MutableStateFlow`? `ReportScreenState` should be sufficient. IMO, for simplicity, we need to have only one state per screen.